core/foundation/test/testStringUtils.cxx

```diff
@@ -20,4 +20,37 @@ TEST(StringUtils, Split)
    test(",,,", {}, true);
    test(",,a", {"", "", "a"}, false);
    test("", {""}, false);
+
+}
+
+TEST(StringUtils, Join)
+{
+   // Test that ROOT::Join behaves like str.join from Python.
+
+   auto test = [](const std::string& ref, const std::string& sep, const std::vector<std::string>& strings) {
+      auto out = ROOT::Join(sep, strings);
+      EXPECT_EQ(out, ref) << "ROOT::Join gave wrong result.";
+   };
+   test("apple,orange,banana", ",", {"apple", "orange", "banana"});
+   test("apple.orange.banana", ".", {"apple", "orange", "banana"});
+   test("apple::orange::banana", "::", {"apple", "orange", "banana"});
+   test("appleorangebanana", "", {"apple", "orange", "banana"});
+   test("apple,,banana", ",", {"apple", "", "banana"});
+   test("apple", ",", {"apple"});
+   test("", ",", {""});
+   test("", "", {""});
+   test("", "", {});
+   test("", ";;", {""});
+
+   EXPECT_EQ("apple,orange,banana", ROOT::Join(",", "apple", "orange", "banana"));
+   EXPECT_EQ("apple.orange.banana", ROOT::Join(".", "apple", "orange", "banana"));
+   EXPECT_EQ("apple::orange::banana", ROOT::Join("::", "apple", "orange", "banana"));
+   EXPECT_EQ("appleorangebanana", ROOT::Join("", "apple", "orange", "banana"));
+   EXPECT_EQ("apple,,banana", ROOT::Join(",", "apple", "", "banana"));
+   EXPECT_EQ("apple", ROOT::Join(",", "apple"));
+   EXPECT_EQ("", ROOT::Join(",", ""));
+   EXPECT_EQ("", ROOT::Join("", ""));
+   EXPECT_EQ("", ROOT::Join(""));
+   EXPECT_EQ("", ROOT::Join(";;", ""));
```

Comment:
Probably unnecessary, unless you can come up with a good reason to keep the variadic template version.